apis/v1alpha1/tcproute_types.go

```diff
@@ -97,6 +97,11 @@ type TCPRouteMatch struct {
 	//
 	// +optional
 	ExtensionRef *RouteMatchExtensionObjectReference `json:"extensionRef" protobuf:"bytes,1,opt,name=extensionRef"`
+	// SNI is the name to match based on TLS Server Name Indication
+	// ([RFC6066](https://tools.ietf.org/html/rfc6066)).
+	//
+	// +optional
+	SNI *string `json:"sni" protobuf:"bytes,2,opt,name=sni"`
```

Comment:
IMO SNI should not be part of a TCPRoute. Routes should contain matchers that are applicable to the protocol. For example, TCPRoute should match based on TCP header information.

Reply:
Maybe ^ is a use case for a TLSRoute that contains a SNI matcher and forwards to a TCPRoute.